src/Bicep.Local.Extension.UnitTests/TypeSpecBuilderTests/TypeSpecGeneratorTests.cs

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Azure.Bicep.Types.Concrete;
+using Azure.Bicep.Types.Index;
+using Bicep.Local.Extension.Host.Attributes;
+using Bicep.Local.Extension.Host.TypeSpecBuilder;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Bicep.Local.Extension.UnitTests.TypeSpecBuilderTests;
+
+[TestClass]
+public class TypeSpecGeneratorTests
+{
+    // Test classes for property type mapping
+    private class TestStringProperty { public string Name { get; set; } = ""; }
+    private class TestBoolProperty { public bool Flag { get; set; } }
+    private class TestIntProperty { public int Count { get; set; } }
+    private class TestUnsupportedProperty { public DateTime When { get; set; } }
+    private class NestedClassProperty { public TestStringProperty Nested { get; set; } = new(); }
+    private class MultipleStringProperties
+    {
+        public string Name { get; set; } = "";
+        public string Description { get; set; } = "";
+    }
+
+    private TypeSettings typeSettings = new(
+        name: "TestSettings",
+        version: "2025-01-01",
+        isSingleton: true,
+        configurationType: new Azure.Bicep.Types.CrossFileTypeReference("index.json", 0));
+    private TypeFactory typeFactory = new([]);
+    private MockRepository mockRepository = null!;
+    private Mock<ITypeProvider> typeProviderMock = null!;
+
+    [TestInitialize]
+    public void Setup()
+    {
+        mockRepository = new MockRepository(MockBehavior.Strict);
+        typeProviderMock = mockRepository.Create<ITypeProvider>();
```

Comment:
We don't need to use `MockRepository` here since we're not verifying multiple mocks at once. The Bicep repo has a `StrictMock<T>.Of<T>()` helper that you can use to create strict mocks.